pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -97,6 +97,11 @@ func (s *AsyncAgentStore) Store(key AgentCommand, value interface{}) {
 			domainInfo.FSFreezeStatus = &status
 		}
 
+		// The event should always have the updated interfaces, otherwise, the vmi status report  may be broken
+		if domainInfo.Interfaces == nil {
```

Comment:
@davidvossel why originally only the updated part is stored on the domainInfo?
AFAIU, this eventually causes the [domainInformer](https://github.com/kubevirt/kubevirt/blob/9c4b6e4101f0d9a82c2f505bd5402c6282fcc880/cmd/virt-handler/virt-handler.go#L263) in virt-handler to contain incomplete domain. Isn't it problematic not only for interfaces?

Reply:
> why originally only the updated part is stored on the domainInfo?

I don't know. This was probably a mistake from the beginning. This is an area of code I often find suspicious.

The logic is trying to indicate on the event the field in the domain info that changed using a `key`, but the domainInfo object is interpreted without looking at the key at all.

I don't see a good reason to only send partial information over the event channel.